File: Web/src/api/ctfTeamUser.js
import {apiClient, handleResponse, handleError} from "./api.js"

function GetCTFTeamUserList(data) {
  return apiClient.post("/ctfTeamUser/GetCTFTeamUserList", data).then(handleResponse).catch(handleError)
}

function AddCTFTeamUser(data) {
    return apiClient.post("/ctfTeamUser/AddCTFTeamUser", data).then(handleResponse).catch(handleError)
}

function RemoveCTFTeamUser(data) {
    return apiClient.post("/ctfTeamUser/RemoveCTFTeamUser", data).then(handleResponse).catch(handleError)
}

const ctfTeamUserApi = {
  GetCTFTeamUserList,
  AddCTFTeamUser,
  RemoveCTFTeamUser
}

export default ctfTeamUserApi

File: Server/route/ctfTeamUser.go
package route

import (
	"github.com/CoolCoolTomato/MatrilxArena/Server/api"
	"github.com/CoolCoolTomato/MatrilxArena/Server/middleware"
	"github.com/gin-gonic/gin"
)

func SetCTFTeamUserRoute(r *gin.Engine) {
	adminCTFTeamUser := r.Group("/ctfTeamUser")
	adminCTFTeamUser.Use(middleware.AdminAuthMiddleware())
	{
		adminCTFTeamUser.POST("/GetCTFTeamUserList", api.GetCTFTeamUserList)
		adminCTFTeamUser.POST("/AddCTFTeamUser", api.AddCTFTeamUser)
		adminCTFTeamUser.POST("/RemoveCTFTeamUser", api.RemoveCTFTeamUser)
	}
}
